extension/wallet/src/protocol/extract_loan.rs

```diff
@@ -1,16 +1,35 @@
-use crate::{
-    storage::Storage,
-    wallet::{compute_balances, current, get_txouts, Wallet},
-    LoanDetails, BTC_ASSET_ID, USDT_ASSET_ID,
-};
+use crate::esplora::EsploraClient;
+use crate::storage::Storage;
+use crate::wallet::current;
+use crate::{BTC_ASSET_ID, ESPLORA_API_URL, LOADED_WALLET, USDT_ASSET_ID};
 use baru::loan::{Borrower0, LoanResponse};
-use elements::secp256k1_zkp::SECP256K1;
-use futures::lock::Mutex;
-use wasm_bindgen::UnwrapThrowExt;
+use baru_wallet::LoanDetails;
+use elements::bitcoin::secp256k1::SECP256K1;
+use wasm_bindgen::prelude::*;
 
-pub async fn extract_loan(
-    name: String,
-    current_wallet: &Mutex<Option<Wallet>>,
+/// Decomposes a loan into:
+///
+/// - Collateral amount, collateral asset balance before and collateral asset balance after.
+/// - Principal amount, principal asset balance before and principal asset balance after.
+/// - Principal repayment amount.
+/// - Loan term.
+///
+/// To do so we unblind confidential `TxOut`s whenever necessary.
+///
+/// This also updates the state of the current loan protocol
+/// "handshake" so that we can later on sign the loan transaction and
+/// give it back to the lender.
+#[wasm_bindgen]
+pub async fn extract_loan(wallet_name: String, loan_response: JsValue) -> Result<JsValue, JsValue> {
+    let loan_response = map_err_from_anyhow!(loan_response.into_serde())?;
+    let loan_details = map_err_from_anyhow!(extract_loan_w(wallet_name, loan_response).await)?;
+    let details = map_err_from_anyhow!(JsValue::from_serde(&loan_details))?;
+
+    Ok(details)
+}
```

Comment:
Does this work if the `wasm_bindgen` annotated function is not at the root of the library?

Reply:
not sure, was hoping someone else would know the answer to this

Reply:
Well, if the tests still run, we should be good :)